packages/core/content-manager/server/src/homepage/services/homepage.ts

```diff
@@ -147,72 +160,68 @@ const createHomepageService = ({ strapi }: { strapi: Core.Strapi }) => {
       model: uid,
     });
 
+  const queryLastDocuments = async (additionalQueryParams?: Record<string, unknown>) => {
+    const permittedContentTypes = await getPermittedContentTypes();
+    const allowedContentTypeUids = permittedContentTypes.filter((uid) => {
+      return contentTypes.hasDraftAndPublish(strapi.contentType(uid));
+    });
+    // Fetch the configuration for each content type in a single query
+    const configurations = await getConfiguration(allowedContentTypeUids);
+    // Get the necessary metadata for the documents
+    const contentTypesMeta = getContentTypesMeta(allowedContentTypeUids, configurations);
+
+    const recentDocuments = await Promise.all(
+      contentTypesMeta.map(async (meta) => {
+        const permissionQuery = await getPermissionChecker(meta.uid).sanitizedQuery.read({
+          limit: MAX_DOCUMENTS,
+          sort: 'publishedAt:desc',
+          fields: meta.fields,
+          ...additionalQueryParams,
+        });
+
+        const docs = await strapi.documents(meta.uid).findMany(permissionQuery);
+        const populate = additionalQueryParams?.populate as string[];
+
+        return formatDocuments(docs, meta, populate);
+      })
+    );
+
+    return recentDocuments
+      .flat()
+      .sort((a, b) => {
+        if (!a.publishedAt || !b.publishedAt) return 0;
+        return b.publishedAt.valueOf() - a.publishedAt.valueOf();
+      })
+      .slice(0, MAX_DOCUMENTS);
+  };
+
   return {
+    async getRecentlyAssignedDocuments(): Promise<GetRecentDocuments.Response['data']> {
```

Comment:
Then this could be moved to review-workflows using the queryLastDocument function on the content-manager homepage service